server/src/test/scripts/data_demo.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import division
+
+import math
+import random
+import time
+
+import labrad
+
+
+def demo_1d_simple(dv):
+    dv.new('demo_1d_simple', ['x'], ['y'])
+    for i in xrange(1000):
+        x = i / 100
+        y = 5 * math.sin(x) + random.random()
+        dv.add([x, y])
+        time.sleep(0.01)
+
+
+def demo_1d_multi(dv):
+    dv.new('demo_1d_multi', ['x'], ['y1', 'y2', 'y3', 'y4', 'y5', 'y6'])
+    for i in xrange(1000):
+        x = i / 100
+        row = [x]
+        for i in range(6):
+            y = 5 * math.sin(x + math.pi * i/5) + random.random()
+            row.append(y)
+        dv.add(row)
+        time.sleep(0.01)
+
+
+def demo_2d_simple(dv):
+    dv.new('demo_2d_simple', ['x', 'y'], ['z'])
+    for i in xrange(50):
+        x = i / 10
+        for j in xrange(50):
+            y = j / 10
+            z = 5 * math.cos((x**2 + y**2) / 10) + random.random()
+            dv.add([x, y, z])
+            time.sleep(0.01)
+
+
+def demo_2d_vargrid(dv):
+    dv.new('demo_2d_vargrid', ['x', 'y'], ['z'])
+    nx, ny = 50, 50
+    dx = [1 + 5 * math.sin(3*math.pi * i / nx)**2 for i in range(nx)]
+    dy = [1 + 5 * math.sin(2*math.pi * i / ny)**2 for i in range(ny)]
+    rMax2 = sum(dx[:-1])**2 + sum(dy[:-1])**2
+    for i in xrange(nx):
+        x = sum(dx[:i])
+        for j in xrange(ny):
+            y = sum(dy[:j])
+            z = 5 * math.cos(4*math.pi * (x**2 + y**2) / rMax2) + random.random()
+            dv.add([x, y, z])
+            time.sleep(0.01)
+
+
+
```

Comment:
I probably should know this but what are the whitespace guidelines between functions? There are 3 spaces here, the rest have 2.